x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/edit_package_policy_page/index.tsx

```diff
@@ -117,6 +117,44 @@ export const EditPackagePolicyForm = memo<{
 
   const canWriteIntegrationPolicies = useAuthz().integrations.writeIntegrationPolicies;
 
+  const newSecrets = useMemo(() => {
```

Comment:
I think it would be great to move that to an util function out of that already huge file, so we can better separate/unit test that logic what do you think?

Reply:
Done in https://github.com/elastic/kibana/pull/171405/commits/286b223ecb0cb5251ec5c9fa9de0acca26517fb8